### PR TITLE
Service call changes

### PR DESCRIPTION
**Pull request by nbinford-formant**

### CODE REVIEW

File: formant_ros2_adapter/scripts/main.py [hunk 1]
```diff
@@ -1189,202 +1193,299 @@ def handle_formant_teleop_msg(self, msg):
...
+        # Handle service calls
+        elif stream_name in self.ros2_service_clients:
+            print(f"INFO: Calling service {stream_name}")
+            for service_client in self.ros2_service_clients[stream_name]:
+                # Get parameter from bitset (button) or numeric (slider/dial) stream
+                # Other teleop stream types are not supported
+                command_text = ""
+                if msg.HasField("bitset"):
+                    if msg.bitset.bits[0].value is True:
+                        command_text = "True"
```
Comment (tanim-islm) [inline formant_ros2_adapter/scripts/main.py:1205]: How come we store the value as a string if its a boolean/bitset?

File: formant_ros2_adapter/scripts/main.py [hunk 2]
```diff
@@ -1189,202 +1193,299 @@ def handle_formant_teleop_msg(self, msg):
...
+            return success, service_result
+
+        # Check if the service is available
+        if service_client.wait_for_service(SERVICE_CALL_TIMEOUT) == False:
+            service_result = ("WARNING: Timeout waiting for service")
+            print(service_result)
+            return success, service_result
+        # Call the service if the paramaters are valid
+        if service_result == "":
+            service_result = service_client.call(service_request)
```
Comment (tanim-islm) [inline formant_ros2_adapter/scripts/main.py:1432]: service_client.call or service_client.call_async?

**Review by tanim-islm — COMMENTED** [review]
Looks all good, only thing is regarding the comment I made. Not sure if it was intentional that its no longer call_async or was just a mistake

File: formant_ros2_adapter/scripts/main.py (hunk 2 above)
Comment (nbinford-formant) [reply formant_ros2_adapter/scripts/main.py:1432]: That was on purpose: we want it to wait for and call the service in a simpler way

**Review by tanim-islm — APPROVED** [review]

**Pull request merged**